src/Regulator.h

```diff
@@ -40,23 +40,37 @@
 // http://www.emptyloop.com/technotes/A%20tutorial%20on%20Burg's%20method,%20algorithm%20and%20recursion.pdf
 // https://metacpan.org/source/SYP/Algorithm-Burg-0.001/README
 
-#ifndef __POOLUFFER_H__
-#define __POOLUFFER_H__
+#ifndef __REGULATOR_H__
+#define __REGULATOR_H__
+
+#include <math.h>
 
 #include <QDebug>
 #include <QElapsedTimer>
 
 #include "AudioInterface.h"
 #include "RingBuffer.h"
-using std::vector;
-#include <math.h>
+
+//#define GUIBS3
+#ifdef GUIBS3
+#include <QWidget>
+
+#include "herlpergui.h"
+#include "ui_herlpergui.h"
+#endif
 
 class BurgAlgorithm
 {
    public:
     bool classify(double d);
-    void train(vector<long double>& coeffs, const vector<float>& x);
-    void predict(vector<long double>& coeffs, vector<float>& tail);
+    void train(std::vector<long double>& coeffs, const std::vector<float>& x);
+    void predict(std::vector<long double>& coeffs, std::vector<float>& tail);
+
+   private:
+    // the following are class members to minimize heap memory allocations
+    std::vector<long double> Ak;
+    std::vector<long double> f;
+    std::vector<long double> b;
```

Comment:
Those will end up on the heap because the size is unknown. To be totally sure that this ends up on the stack is to use std::array with the maximum possible size. With C++20 std::span would probably be the way to go.

Nevertheless this shouldn't stay in the way of merging as this feature stays experimental.

Reply:
A quick look at this and the solution is going to be non-trivial